testing/dart/platform_isolate_test.dart

```diff
@@ -0,0 +1,97 @@
+// Copyright 2024 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:isolate';
+import 'dart:ui';
+
+import 'package:litetest/litetest.dart';
+
+void main() {
+  test('PlatformIsolate isRunningInPlatformThread, false cases', () async {
+    final bool isPlatThread =
+        await Isolate.run(() => isRunningInPlatformThread());
+    expect(isPlatThread, isFalse);
+  });
+
+  test('PlatformIsolate runInPlatformThread', () async {
+    final bool isPlatThread =
+        await runInPlatformThread(() => isRunningInPlatformThread());
+    expect(isPlatThread, isTrue);
+  });
+
+  test('PlatformIsolate runInPlatformThread, async operations', () async {
+    final bool isPlatThread = await runInPlatformThread(() async {
+      await Future<void>.delayed(const Duration(milliseconds: 100));
+      await Future<void>.delayed(const Duration(milliseconds: 100));
+      await Future<void>.delayed(const Duration(milliseconds: 100));
+      return isRunningInPlatformThread();
+    });
+    expect(isPlatThread, isTrue);
+  });
```

Comment:
> At the moment you can only call runInPlatformThread from the root isolate.

It would be very easy to do here (for helper isolate I understand why it's more work and could be made in future CL).

I've added it as follow-up tasks on the issue (https://github.com/flutter/flutter/issues/136314)